### PR TITLE
  [IGA] Pressure follower load

### PR DESCRIPTION
**Pull request by rickyaristio**
**Description**
Adds the feature to compute the follower loads in IGA where the load direction is always perpendicular with the surface. Thus, there is an extra calculation for the load stiffness matrix.

### CODE REVIEW

**Comment by tteschemacher** [comment]
It is different as discussed, but I am very much in favor of your chosen path.

One thing, I think this follower load is purely for KL shells? May I suggest to rename it accordingly being FOLLOWER_LOAD_KL? what do you think?

Also do you think it would be possible to do a small test, i.e. the rolling example?

**Comment by rickyaristio** [comment]
> It is different as discussed, but I am very much in favor of your chosen path.
> 
> One thing, I think this follower load is purely for KL shells? May I suggest to rename it accordingly being FOLLOWER_LOAD_KL? what do you think?
> 
> Also do you think it would be possible to do a small test, i.e. the rolling example?

This follower load can be applied to all elements. More precisely, this is a pressure follower load. Similar to regular pressure load, it will follow the surface as it deforms. The difference is that this one takes into account the load stiffness matrix (the first-order derivative of the residual forces will not vanish due to the normal vector of the surface).

I've renamed it to PRESSURE_FOLLOWER_LOAD to avoid misunderstanding. 

Regarding the test, I will add it in the future.

**Review by tteschemacher — APPROVED** [review]
Forgot about that PR I thought it was already merged.

File: applications/IgaApplication/custom_conditions/load_condition.cpp [hunk 1]
```diff
@@ -150,6 +150,75 @@ namespace Kratos
...
+
+                    a2x(0,1) = a2[2];
+                    a2x(0,2) = -a2[1];
+                    a2x(1,2) = a2[0];
+                    a2x(1,0) = a2x(0,1);
+                    a2x(2,0) = a2x(0,2);
+                    a2x(2,1) = a2x(1,2);
+
+                    // d. compute the left hand side
+                    noalias(rLeftHandSideMatrix) -= integration_weight * pressure_follower_load * (prod(trans(N), (prod(a2x, DN_DXi) + prod(a1x, DN_DEta))));
```
Comment (tteschemacher) [inline applications/IgaApplication/custom_conditions/load_condition.cpp:219]: are you sure that you are using integration weight here instead of d_weight?

Comment (rickyaristio) [reply applications/IgaApplication/custom_conditions/load_condition.cpp:219]: Yes, the determinant_jacobian has vanished.

`noalias(rLeftHandSideMatrix) -= d_weight * pressure_follower_load * (prod(trans(N), (prod(a2x, DN_DXi) + prod(a1x, DN_DEta)))) / determinat_jacobian_vector[point_number];`

`noalias(rLeftHandSideMatrix) -= integration_weight * pressure_follower_load * (prod(trans(N), (prod(a2x, DN_DXi) + prod(a1x, DN_DEta))));`

**Comment by rickyaristio** [comment]
> Forgot about that PR I thought it was already merged.

Thanks for the approval!

**Pull request merged**